Processing_data/Classes/StatsBucket.cs

```diff
@@ -0,0 +1,52 @@
+namespace Processing_data.Classes
+{
+    /// <summary>
+    /// Represents a collection of statistics for a set of data
+    /// </summary>
+    public class StatsBucket : Dictionary<Coordinate, Statistics>
+    {
+        /// <summary>
+        /// Adds a data value to the statistics bucket
+        /// </summary>
+        /// <param name="coordinate"></param>
+        /// <param name="data"></param>
+        public void AddData(Coordinate coordinate, double data)
+        {
+            if (!ContainsKey(coordinate))
+            {
+                Add(coordinate, new Statistics(new List<double> { data }));
+            }
+            else
+            {
+                this[coordinate].AddData(data);
+            }
+        }
+
+        public StatsBucket() : base(new CustomEqualityComparer())
+        {
+        }
+
+        /// <summary>
+        /// Prints the statistics for each coordinate in the bucket
+        /// </summary>
+        public void PrintStats()
```

Comment:
this looks like a presentation logic, which deserves a dedicated class.

Reply:
I made changes.
I added new class which print stats

Reply:
The last step is to remove code here.